code/modules/projectiles/guns/ballistic/revolver.dm

```diff
@@ -210,94 +236,139 @@
+	if(!spun)
+		to_chat(user, span_warning("You need to spin \the [src]'s chamber first!"))
+		return TRUE // no melee attack
+	if(HAS_TRAIT(user, TRAIT_CURSED)) // I cannot live, I cannot die, trapped in myself, body my holding cell.
+		to_chat(user, span_warning("What a horrible night... To have a curse!"))
+		return TRUE // no melee attack
+	if(loc != user)
+		if(tk_firing(user))
+			to_chat(user, span_warning("Russian roulette is stressful enough without trying to focus on telekinesis!"))
+		else
+			to_chat(user, span_warning("You need to be holding the gun to fire it!"))
+		return TRUE // no melee attack
+
+	return ..() // try to shoot the gun
+
+// Replaces clumsy check with a do after
+/obj/item/gun/ballistic/revolver/russian/check_botched(mob/living/user, atom/target)
+	if(aim_time <= 0)
+		return FALSE
+	user.visible_message(
+		span_danger("[user] aims \the [src] at [user.p_their()] [parse_zone(user.zone_selected)]..."),
+		span_userdanger("You aim \the [src] at your [parse_zone(user.zone_selected)]..."),
+		visible_message_flags = ALWAYS_SHOW_SELF_MESSAGE,
+	)
+	if(prob(10)) // future idea : make badasses or insane people not jitter here
+		user.adjust_jitter(aim_time)
+	if(!do_after(user, aim_time, target))
+		if(!user.incapacitated)
+			user.visible_message(
+				span_danger("[user] loses [user.p_their()] nerve and puts \the [src] down."),
+				span_userdanger("You lose your nerve and put \the [src] down."),
+				visible_message_flags = ALWAYS_SHOW_SELF_MESSAGE,
 			)
+		return TRUE
+	return FALSE
 
-		if(chambered)
-			var/obj/item/ammo_casing/AC = chambered
-			if(AC.fire_casing(user, user, params, distro = 0, quiet = 0, zone_override = null, spread = 0, fired_from = src))
-				playsound(user, fire_sound, fire_sound_volume, vary_fire_sound)
-				if(is_target_face)
-					shoot_self(user, affecting)
-				else if(tk_controlled) // the consequence of you doing the telekinesis stuff
-					to_chat(user, span_userdanger("As your mind concentrates on the revolver, you realize that it's pointing towards your head a little too late!"))
-					shoot_self(user, BODY_ZONE_HEAD)
-				else
-					user.visible_message(span_danger("[user.name] cowardly fires [src] at [user.p_their()] [affecting.name]!"), span_userdanger("You cowardly fire [src] at your [affecting.name]!"), span_hear("You hear a gunshot!"))
-				chambered = null
-				user.add_mood_event("russian_roulette_lose", /datum/mood_event/russian_roulette_lose)
-				return
+/obj/item/gun/ballistic/revolver/russian/before_firing(atom/target, mob/user)
+	if(target != user)
+		CRASH("Russian revolver somehow got to before_firing with a target that isn't the user!")
+	// we will definitely have a chambered round, but not always projectile
+	if(check_zone(user.zone_selected) == BODY_ZONE_HEAD)
+		chambered.loaded_projectile?.damage = 300
+		chambered.loaded_projectile?.wound_bonus = 100
+	else
+		chambered.loaded_projectile?.damage = 80
+		chambered.loaded_projectile?.wound_bonus = 10
```

Comment:
if the bullet fails to fire, will it keep the damage modifications made here?

Reply:
As far as I'm aware at this point it cannot fail to fire